geth/jail/internal/loop/loop.go

```diff
@@ -54,6 +77,18 @@ func NewWithBacklog(vm *vm.VM, backlog int) *Loop {
 		vm:    vm,
 		tasks: make(map[int64]Task),
 		ready: make(chan Task, backlog),
+		state: state{
+			accept: true,
+		},
+	}
+}
+
+func (l *Loop) close() {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	if l.accepting() {
```

Comment:
Hmm... Does it works?
```
func (s *state) accepting() bool {
	s.lock.Lock()
	defer s.lock.Unlock()
	return s.accept
}
```
So when we call `if l.accepting()` we'll try to lock already locked mutes. It looks like deadlock. Could you check it?

Reply:
The mutex from `loop` and the mutex from `state` are different, so each needs to be locked individually.